File: Computer-Seekho-.NET/Computer_Seekho_DN/Controllers/AlbumController.cs
using Computer_Seekho_DN.Models;
using Computer_Seekho_DN.Service;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace Computer_Seekho_DN.Controllers;

[Route("album")]
[ApiController]
public class AlbumController : ControllerBase
{
    private readonly IAlbumService _albumService;

    public AlbumController(IAlbumService albumService)
    {
        _albumService = albumService;
    }

    [HttpPost("add")]
    public async Task<IActionResult> CreateAlbum([FromBody] Album album)
    {
        if(album == null) { return BadRequest(new { message = "Invalid Details" }); }
        var savedAlbum = await _albumService.SaveAlbumAsync(album);
        return Ok(new {message = "Album Added"});
    }

    [HttpGet("getAll")]
    public async Task<IActionResult> GetAllAlbums()
    {
        var albums = await _albumService.GetAllAlbumsAsync();
        return Ok(albums);
    }

    [HttpGet("get/{id}")]
    public async Task<IActionResult> GetAlbumById(int id)
    {
        var album = await _albumService.GetAlbumByIdAsync(id);
        return album != null ? Ok(album) : NotFound();
    }

    [HttpPut("update/{id}")]
    public async Task<IActionResult> UpdateAlbum(int id, [FromBody] Album albumDetails)
    {
        bool isUpdated = await _albumService.UpdateAlbumAsync(albumDetails);
        return isUpdated ? Ok("Album updated successfully") : NotFound("Album not found");
    }

    [HttpDelete("delete/{id}")]
    public async Task<IActionResult> DeleteAlbum(int id)
    {
        bool isDeleted = await _albumService.DeleteAlbumAsync(id);
        return isDeleted ? Ok(new { message = "Deleted Successfully" }) : NotFound(new { message = "Album not found" });
    }
}

File: Computer-Seekho-Java/ComputerSeekho/src/main/java/com/Project/Controllers/AlbumController.java
package com.Project.Controllers;

import com.Project.DTO.ApiResponse;
import com.Project.Entities.Album;
import com.Project.Services.AlbumService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.time.LocalDateTime;
import java.util.List;

@RestController
@RequestMapping("/albums")
public class AlbumController {

    @Autowired
    private AlbumService albumService;

    @PostMapping("/add")
    public ResponseEntity<Album> createAlbum(@RequestBody Album album) {
        Album savedAlbum = albumService.saveAlbum(album);
        return new ResponseEntity<>(savedAlbum, HttpStatus.CREATED);
    }

    @GetMapping("/getAll")
    public ResponseEntity<List<Album>> getAllAlbums() {
        List<Album> albums = albumService.getAllAlbums();
        return new ResponseEntity<>(albums, HttpStatus.OK);
    }

    @GetMapping("/{id}")
    public ResponseEntity<Album> getAlbumById(@PathVariable("id") int albumId) {
        Album album = albumService.getAlbumById(albumId);
        return album != null ? new ResponseEntity<>(album, HttpStatus.OK) : new ResponseEntity<>(HttpStatus.NOT_FOUND);
    }

    @PutMapping("/update/{id}")
    public ResponseEntity<ApiResponse> updateAlbum(@PathVariable("id") int albumId, @RequestBody Album albumDetails) {
        if(albumService.updateAlbum(albumDetails)){
            return new ResponseEntity<>(new ApiResponse("Album updated successfully",LocalDateTime.now()), HttpStatus.OK);
        } else {
            return new ResponseEntity<>(new ApiResponse("Album not found",LocalDateTime.now()), HttpStatus.NOT_FOUND);
        }
    }

    @DeleteMapping("/delete/{id}")
    public ResponseEntity<ApiResponse> deleteAlbum(@PathVariable("id") int albumId) {
        albumService.deleteAlbum(albumId);
        return new ResponseEntity<>(new ApiResponse("Deleted Successfully",LocalDateTime.now()),HttpStatus.OK);
    }
}